omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_ipv6_validator.rb

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative './preflight_checks.rb'
+
+class Ipv6PreflightValidator < PreflightValidator
+  def initialize(node)
+    super
+  end
+
+  def run!
+    verify_ipv6_for_lo
+  end
+
+  def verify_ipv6_for_lo
+    if ipv6_enabled && !lo_has_ipv6_addr
+      fail_with <<EOF
+Your system has IPv6 enabled but its loopback interface has no IPv6
+address.
+
+You must either pass `ipv6.disable=1` to your kernel command line,
+or ensure the loopback interface has an `::1` address by running
+
+    sysctl net.ipv6.conf.lo.disable_ipv6=0
+
+or
+
+    echo 0 > /proc/sys/net/ipv6/conf/lo/disable_ipv6
```

Comment:
Since both of these do the same thing, I think we should just offer on set of instructions.  I'm partial to `sysctl` because it is more easily prepended with `sudo`